File: src/js/next/src/Style/MetaLicense.ts
import type { StyleDefinitionMetaLicense } from '../StyleDefinition.js';


export class MetaLicense {
  #data: StyleDefinitionMetaLicense;

  constructor(data: StyleDefinitionMetaLicense) {
    this.#data = data;
  }

  name(): string | undefined {
    return this.#data.name;
  }

  url(): string | undefined {
    return this.#data.url;
  }

  text(): string | undefined {
    return this.#data.text;
  }
}

File: src/go/core/internal/style/meta_license.go
package style

// MetaLicense is a view over the meta.license block of a style definition.
//
// The fields are *string so the license builder can distinguish an absent field
// from one explicitly set to the empty string, matching the JS nullish
// (`?? 'Unknown'`) and Rust Option logic.
type MetaLicense struct {
	Name *string `json:"name"`
	URL  *string `json:"url"`
	Text *string `json:"text"`
}
